cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/local/LocalStorageClient.java

```diff
@@ -0,0 +1,39 @@
+package com.linkedin.openhouse.cluster.storage.local;
+
+import com.linkedin.openhouse.cluster.storage.StorageClient;
+import com.linkedin.openhouse.cluster.storage.StorageType;
+import java.io.IOException;
+import javax.annotation.PostConstruct;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+/**
+ * The LocalStorageClient class is an implementation of the StorageClient interface for local
+ * storage. It uses an Apache Hadoop FileSystem to interact with the local file system.
+ */
+@Slf4j
+@Lazy
+@Component
+public class LocalStorageClient implements StorageClient<FileSystem> {
+
+  private FileSystem fs;
+
+  /**
+   * Initialize the LocalStorageClient when the bean is accessed for the first time.
+   *
+   * @throws IOException if an I/O error occurs
+   */
+  @PostConstruct
+  private synchronized void init() throws IOException {
+    log.info("Initialized storage client for type: " + StorageType.LOCAL.getValue());
+    this.fs = FileSystem.get(new Configuration());
```

Comment:
You might want to do a check here to verify that the FileSystem instance returned is an instanceOf LocalFileSystem.

Depending on what is in the classpath, this call could return a FileSystem impl that is not local. It might be good to guarantee that this is the local filesystem for this class since it is called LocalStorageClient.

Another option that might even be stronger is to change the type parameter for the class to be LocalFileSystem instead of FileSystem.

Reply:
great feedback! we don't need to initialize LocalStorage  based on classpath/configuration. Changed to instantiate purely `LocalFileSystem`